src/shell_executor.rs

```diff
@@ -78,19 +80,23 @@ network={{
         info!("Attempting to connect with supplicant:\n{}\n", supplicant);
         fs::write("/etc/wpa_supplicant/wpa_supplicant.conf", supplicant)?;
 
-        let daemon_reload = self.execute("systemctl", &["daemon-reload"])?;
-        if !daemon_reload.success() {
-            error!("Failed to systemctl daemon reload");
-            return Ok(daemon_reload);
-        }
+        //let daemon_reload = self.execute("systemctl", &["daemon-reload"])?;
+        //if !daemon_reload.success() {
+        //    error!("Failed to systemctl daemon reload");
+        //    return Ok(daemon_reload);
+        //}
+
+        //self.execute("sudo", &["dhclient", "-r", "wlan0"])?;
 
-        self.execute("sudo", &["dhclient", "-r", "wlan0"])?;
+        self.set_interface("wlan0", true);
 
-        self.execute("sudo", &["ifdown", "wlan0"])?;
+        let output = self.execute("sudo", &["systemctl", "restart", "wpa_supplicant.service"])?;
 
-        self.execute("sudo", &["ifup", "wlan0"])?;
+        info!("Sleeping for a little bit to allow wpa to catch up after being restarted.");
+        sleep(Duration::from_secs(15));
```

Comment:
Necessary to sleep because the `hostname -I` call to show the sync code will fail if run too soon.